src/Storage/Storage/Common/Cmdlet/NewAzureStorageContext.cs

```diff
@@ -462,6 +482,29 @@ private string GetTokenStrFromAccessToken(IAccessToken accessToken)
             return new CloudStorageAccount(credential, new Uri(blobEndpoint), new Uri(queueEndpoint), new Uri(tableEndpoint), new Uri(fileEndpoint));
         }
 
+        /// <summary>
+        /// Get storage account and use specific end point
+        /// </summary>
+        /// <param name="credential">Storage credential</param>
+        /// <param name="storageAccountName">Storage account name, it's used for build end point</param>
+        /// <param name="useHttps"></param>
+        /// <param name="endPoint"></param>
+        /// <returns>A storage account</returns>
+        internal CloudStorageAccount GetStorageAccountWithCustomEndPoint(StorageCredentials credential, bool useHttps, string endPoint)
+        {
+            if (string.IsNullOrEmpty(endPoint))
+            {
+                throw new ArgumentException(String.Format(Resources.ObjectCannotBeNull, "CustomEndpoint"));
+            }
+
+            string blobEndpoint = endPoint;
+            string tableEndpoint = endPoint;
+            string queueEndpoint = endPoint;
+            string fileEndpoint = endPoint;
```

Comment:
It's really odd to use same endpoint for blob/file/queue/table. Not sure which scenario will use it.

Reply:
We should allow user to input different endpoint for different service.
It's really odd that B/T/Q/F all use same endpoint.